src/depot_gui.cpp

```diff
@@ -886,13 +885,13 @@ struct DepotWindow : Window {
 			SetDParam(1, loaded[cargo_type]);   // {CARGO} #2
 			SetDParam(2, cargo_type);           // {SHORTCARGO} #1
 			SetDParam(3, capacity[cargo_type]); // {SHORTCARGO} #2
-			pos = GetString(pos, STR_DEPOT_VEHICLE_TOOLTIP_CARGO, lastof(details));
+			details += GetString(STR_DEPOT_VEHICLE_TOOLTIP_CARGO);
 		}
 
 		/* Show tooltip window */
 		uint64 args[2];
 		args[0] = (whole_chain ? num : v->engine_type);
-		args[1] = (uint64)(size_t)details;
+		args[1] = (uint64)(size_t)details.c_str();
```

Comment:
wauw, I needed a moment to process this cast-hell happening. I don't even .. let's continue on :)

(nothing you have to do; just weird code before your PR)

Reply:
Yes, solving this is something else... though something that eventually needs to be done. However, with `std::string` that might be a bit easier to accomplish.